slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -19,9 +19,45 @@ std::atomic<bool> b_continue_session{true};
 ::grpc::Status SLAMServiceImpl::GetPosition(ServerContext *context,
                                             const GetPositionRequest *request,
                                             GetPositionResponse *response) {
-    LOG(ERROR) << "GetPosition is not yet implemented.\n";
-    return grpc::Status(grpc::StatusCode::UNIMPLEMENTED,
-                        "GetPosition is not yet implemented.");
+
+    cartographer::transform::Rigid3d global_pose;
+    {
+        std::lock_guard<std::mutex> lk(map_builder_mutex);
+        if (map_builder.GetLocalSlamResultPoses().size() > 0) {
+            global_pose = map_builder.GetLastGlobalPose(trajectory_id);
+            LOG(INFO) << "global_pose: " << global_pose.DebugString();
+        } else {
+            LOG(ERROR) << "GetPosition is not ready.\n";
+            return grpc::Status(grpc::StatusCode::UNAVAILABLE,
+                                "GetPosition is not ready yet");
+        }
+    }
+    // Setup mapping of pose message to the response. NOTE not using
+    // inFrame->set_reference_frame yet
```

Comment:
[q] When are we going to use it? Is this going to be changed in a future ticket?

Reply:
[q] Curious if this comment should have an associated ticket with it, so we can provide more information and/or know when this will be changed?

Reply:
Currently we do not plan to use the reference frame part of our pose message. When we redo our [proto methods](https://viam.atlassian.net/browse/DATA-431) in the map representation tickets the reference frame of the proto will be phased out. I think we should still note the reference frame is not being used(can remove the yet)